pkg/broker/ingress/multi_topic_decouple_sink.go

```diff
@@ -90,6 +103,36 @@ func (m *multiTopicDecoupleSink) Send(ctx context.Context, broker types.Namespac
 	return err
 }
 
+// EventFilterFunc is used to see if a target is interested in an event.
+// It is used as a vaiable to allow stubbing out in unit tests.
+var EventFilterFunc = filter.PassFilter
+
+// EnableEventFilterFunc is a temporary function to control enabling and
+// disabling trigger-less event filtering in ingress.
+// TODO(#1804): remove this variable when enabling the feature by default.
+var EnableEventFilterFunc = isEventFilteringEnabled
+
+// TODO(#1804): remove this method when enabling the feature by default.
+func isEventFilteringEnabled() bool {
+	return os.Getenv("ENABLE_EVENT_FILTERING") == "true"
```

Comment:
How will this be set?

IIUC, this Deployment is created by the BrokerCell reconciler. How will that reconciler choose to set or not set this env var?

Reply:
Good point. I have added annotations to the broker cell now which will set this env variable on ingress pods if they are set.